gammagl/layers/conv/hardgat_conv.py

```diff
@@ -0,0 +1,173 @@
+        attention coefficients which exposes each node to a stochastically
+        sampled neighborhood during training. (default: :obj:`0`)
+    add_self_loops: bool, optional
+        If set to :obj:`False`, will not add
+        self-loops to the input graph. (default: :obj:`True`)
+    add_bias: bool, optional
+        If set to :obj:`False`, the layer will not learn
+        an additive bias. (default: :obj:`True`)
+
+    """
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 k=8,
+                 heads=1,
+                 concat=True,
+                 negative_slope=0.2,
+                 dropout_rate=0,
+                 add_bias=True):
+        super().__init__()
+
+        self.in_channels = in_channels
+        self.out_channels = out_channels
+        self.heads = heads
+        self.concat = concat
+        self.negetive_slop = negative_slope
+        self.dropout_rate = dropout_rate
+        self.k = k
+        # self.add_self_loops = add_self_loops
+        self.add_bias = add_bias
+
+        self.linear = tlx.layers.Linear(out_features=self.out_channels * self.heads,
+                                        in_features=self.in_channels,
+                                        b_init=None)
+        initor = tlx.initializers.XavierNormal(gain=math.sqrt(2.0))
+        self.p = self._get_weights("proj", shape=(1, self.in_channels), init=initor, order=True)
+        initor = tlx.initializers.TruncatedNormal()
+        self.att_src = self._get_weights("att_src", shape=(1, self.heads, self.out_channels), init=initor, order=True)
+        self.att_dst = self._get_weights("att_dst", shape=(1, self.heads, self.out_channels), init=initor, order=True)
+
+        self.leaky_relu = tlx.layers.LeakyReLU(negative_slope)
+        self.dropout = tlx.layers.Dropout(self.dropout_rate)
+
+        if self.add_bias and concat:
+            self.bias = self._get_weights("bias", shape=(self.heads * self.out_channels,), init=initor)
+        elif self.add_bias and not concat:
+            self.bias = self._get_weights("bias", shape=(self.out_channels,), init=initor)
+
+    def select_topk(self, edge_index, value):
+        src, dst = tlx.convert_to_numpy(edge_index)
+        score = tlx.convert_to_numpy(value)
+        score = score[src]
+
+        # 根据value进行降序排列
+        rank = np.argsort(score)[::-1]
+        src = src[rank]
+        dst = dst[rank]
+        # 根据dst进行升序排列
+        index = np.argsort(dst)
+        src = src[index]
+        dst = dst[index]
+
+        # 每个dst节点筛选前k个边
+        e_id = []
+        rowptr = np.concatenate(([0], np.bincount(dst).cumsum()))
+        for dst_node in np.unique(dst):
+            st = rowptr[dst_node]
+            len1 = rowptr[dst_node + 1] - st
+            for i in range(st, min(st + self.k, st + len1)):
+                e_id.append(i)
+
+        return tlx.convert_to_tensor(np.array([src[e_id], dst[e_id]]))
```

Comment:
The sort you used in this function is ``np.argsort``. Could you refer to this link: https://tensorlayerx.readthedocs.io/en/latest/modules/files.html#sort and decide if this can be used.